src/network/networkd-address.c

```diff
@@ -555,6 +562,57 @@ static int address_set_masquerade(Address *address, bool add) {
         return 0;
 }
 
+static void address_modify_nft_set_context(Address *address, bool add, NFTSetContext *nft_set_context) {
+        int r;
+
+        assert(address);
+        assert(address->link);
+        assert(nft_set_context);
+
+        FOREACH_ARRAY(nft_set, nft_set_context->sets, nft_set_context->n_sets) {
+                assert(nft_set);
+
+                r = nft_set_element_modify_in_addr_open(&address->link->manager->fw_ctx, add, nft_set,
+                                                        address->family, &address->in_addr, address->prefixlen);
+                if (r < 0)
+                        log_warning_errno(r, "Failed to %s NFT set: family %s, table %s, set %s, IP address %s, ignoring",
+                                          add? "add" : "delete",
+                                          nfproto_to_string(nft_set->nfproto), nft_set->table, nft_set->set,
+                                          IN_ADDR_PREFIX_TO_STRING(address->family, &address->in_addr, address->prefixlen));
+        }
+}
+
+static void address_modify_nft_set(Address *address, bool add) {
+        assert(address);
+        assert(address->link);
+
+        if (!IN_SET(address->family, AF_INET, AF_INET6))
+                return;
+
+        switch (address->source) {
+        case NETWORK_CONFIG_SOURCE_DHCP4:
+                if (!address->link->network)
```

Comment:
Please move the check to the beginning, even static case, `address->link->network` should be non-NULL.

Reply:
At least assert(address->link->network) won't work, in some cases it _is_ NULL.

Reply:
Yeah, assertion cannot work, as it is also called for addresses on unmanaged interfaces, or on early enumeration.

Reply:
Replaced `assert()` with `if ()`